aws/resource_aws_rds_global_cluster.go

```diff
@@ -0,0 +1,354 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsRDSGlobalCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsRDSGlobalClusterCreate,
+		Read:   resourceAwsRDSGlobalClusterRead,
+		Update: resourceAwsRDSGlobalClusterUpdate,
+		Delete: resourceAwsRDSGlobalClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"database_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"deletion_protection": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"engine": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "aurora",
+				ValidateFunc: validation.StringInSlice([]string{
+					"aurora",
+				}, false),
+			},
+			"engine_version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"global_cluster_identifier": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"global_cluster_resource_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"source_db_cluster_identifier": {
```

Comment:
I don't see this being used anywhere. Should we remove it or was it missed?

Reply:
Good find and probably something I should've mentioned in the commit message/PR comment!

The RDS API gives you this second option for creating Global Clusters, based on an existing Cluster with global engine mode, instead of creating an "empty" Global Cluster. I think there were two reasons for omitting this to start.

When playing with this, the workflow wound up being very tedious because of operation ordering requirements within the RDS API and Terraform references. If I recall correctly, since the Cluster becomes a child of the Global Cluster in the dependency graph and Global Clusters, it wasn't easy to setup the resources to properly destroy. I may have inadvertently fixed this issue with the `aws_rds_cluster` resource `Delete` function update though. 😅 

The other workflow issue is that engine mode cannot be currently updated for existing Clusters, so essentially you have to create new Clusters anyways to get Global Cluster support. I didn't think we needed to introduce all the complexity of supporting both creation methods initially, since operators can just adjust their initial Global Cluster configuration to match our supported workflow. I may have missed something in my thinking there though.

TL;DR I'll remove this attribute for now since its not used. If there are use cases that require it, we can add it (and all its complexity) later. 👍

Reply:
Now that I'm thinking about it more -- it was that because of the referencing (Global Cluster being dependent on Cluster), we'd try to destroy the Global Cluster first and it would complain that it still had members. We'd have to introduce a flag in `aws_rds_global_cluster` to allow you to force remove all members first. Not unsurmountable, but definitely additional complexity which we might not need.

Reply:
Ouch, that extra flag seems tricky but I'm totally onboard with seeing this come in later if it's requested.